### PR TITLE
enables oidc kubeconfig feature/endpoint on api server

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: updates api server deployment to enable `oidc kubeconfg` feature.

**Special notes for your reviewer**: merge blocked by https://github.com/kubermatic/secrets/pull/196

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: config/kubermatic/templates/kubermatic-api-dep.yaml [hunk 1]
```diff
@@ -27,13 +27,19 @@ spec:
           - -logtostderr
           - -datacenters=/opt/datacenter/datacenters.yaml
           - -oidc-url={{ .Values.kubermatic.auth.tokenIssuer }}
-          - -oidc-authenticator-client-id={{ .Values.kubermatic.auth.clientID }}
+          - -oidc-authenticator-client-id={{ .Values.kubermatic.auth.url }}
```
Comment (zreigz) [inline config/kubermatic/templates/kubermatic-api-dep.yaml]: This value looks suspicious `client-id != url`

**Review by zreigz — COMMENTED** [review]
rest looks good

File: config/kubermatic/templates/kubermatic-api-dep.yaml (hunk 1 above)
Comment (p0lyn0mial) [reply config/kubermatic/templates/kubermatic-api-dep.yaml]: yup, just changed it :)

File: config/kubermatic/templates/kubermatic-api-dep.yaml [hunk 2]
```diff
@@ -26,14 +26,20 @@ spec:
           - -v=4
           - -logtostderr
           - -datacenters=/opt/datacenter/datacenters.yaml
-          - -oidc-url={{ .Values.kubermatic.auth.tokenIssuer }}
+          - -oidc-url={{ .Values.kubermatic.auth.url }}
```
Comment (p0lyn0mial) [inline config/kubermatic/templates/kubermatic-api-dep.yaml:29]: I want to rename `tokenIssuer` to `url`

**Review by zreigz — APPROVED** [review]
LGTM

**Pull request merged**